drivers/gpio/gpio_rv32m1.c

```diff
@@ -34,25 +34,72 @@ struct gpio_rv32m1_data {
 	u32_t pin_callback_enables;
 };
 
+static u32_t get_port_pcr_irqc_value_from_flags(struct device *dev,
+		u32_t pin, enum gpio_int_mode mode,
+		enum gpio_int_trig trig)
+{
+	port_interrupt_t port_interrupt = 0;
+
+	if (mode == GPIO_INT_MODE_DISABLED) {
+		port_interrupt = kPORT_InterruptOrDMADisabled;
+	} else {
+		if (mode == GPIO_INT_MODE_LEVEL) {
+			if (trig == GPIO_INT_TRIG_LOW) {
+				port_interrupt = kPORT_InterruptLogicZero;
+			} else {
+				port_interrupt = kPORT_InterruptLogicOne;
+			}
+		} else {
+			switch (trig) {
+			case GPIO_INT_TRIG_LOW:
+				port_interrupt = kPORT_InterruptFallingEdge;
+				break;
+			case GPIO_INT_TRIG_HIGH:
+				port_interrupt = kPORT_InterruptRisingEdge;
+				break;
+			case GPIO_INT_TRIG_BOTH:
+				port_interrupt = kPORT_InterruptEitherEdge;
+				break;
+			}
+		}
+	}
+
+	return PORT_PCR_IRQC(port_interrupt);
+}
+
 static int gpio_rv32m1_configure(struct device *dev,
 				 int access_op, u32_t pin, int flags)
 {
 	const struct gpio_rv32m1_config *config = dev->config->config_info;
 	GPIO_Type *gpio_base = config->gpio_base;
 	PORT_Type *port_base = config->port_base;
-	port_interrupt_t port_interrupt = 0;
+	struct gpio_rv32m1_data *data = dev->driver_data;
 	u32_t mask = 0U;
 	u32_t pcr = 0U;
 	u8_t i;
 
+	/* Check for an invalid pin number */
+	if (pin >= ARRAY_SIZE(port_base->PCR)) {
+		return -EINVAL;
+	}
+
 	/* Check for an invalid pin configuration */
-	if ((flags & GPIO_INT) && (flags & GPIO_DIR_OUT)) {
+	if ((flags & GPIO_INT_ENABLE) && ((flags & GPIO_INPUT) == 0)) {
```

Comment:
We're mixing styles. To be consistent / Misra-C compliant we would need to write it as
```
if (((flags & GPIO_INT_ENABLE) != 0) && ((flags & GPIO_INPUT) == 0)) {
```

Reply:
Actually, I believe we should move this check to `gpio_rv32m1_pin_interrupt_configure` function (and read input/output configuration from the register).